src/webapp/pages/form-page/disease-outbreak-event/utils/diseaseOutbreakEventFormMapper.ts

```diff
@@ -0,0 +1,806 @@
+import { getHazardTypeValue } from "../../../../../data/repositories/consts/DiseaseOutbreakConstants";
+
+export const diseaseOutbreakEventFieldIds = {
+    name: "name",
+    hazardType: "hazardType",
+    mainSyndromeCode: "mainSyndromeCode",
+    suspectedDiseaseCode: "suspectedDiseaseCode",
+    notificationSourceCode: "notificationSourceCode",
+    areasAffectedProvinceIds: "areasAffectedProvinceIds",
+    areasAffectedDistrictIds: "areasAffectedDistrictIds",
+    incidentStatus: "incidentStatus",
+    emergedDate: "emerged_date",
+    emergedNarrative: "emerged_narrative",
+    detectedDate: "detected_date",
+    detectedNarrative: "detected_narrative",
+    notifiedDate: "notified_date",
+    notifiedNarrative: "notified_narrative",
+    initiateInvestigation: "initiateInvestigation",
+    conductEpidemiologicalAnalysis: "conductEpidemiologicalAnalysis",
+    laboratoryConfirmationDate: "laboratoryConfirmation_date",
+    laboratoryConfirmationNA: "laboratoryConfirmation_na",
+    appropriateCaseManagementDate: "appropriateCaseManagement_date",
+    appropriateCaseManagementNA: "appropriateCaseManagement_na",
+    initiatePublicHealthCounterMeasuresDate: "initiatePublicHealthCounterMeasures_date",
+    initiatePublicHealthCounterMeasuresNA: "initiatePublicHealthCounterMeasures_na",
+    initiateRiskCommunicationDate: "initiateRiskCommunication_date",
+    initiateRiskCommunicationNA: "initiateRiskCommunication_na",
+    establishCoordination: "establishCoordination",
+    responseNarrative: "responseNarrative",
+    incidentManagerName: "incidentManagerName",
+    notes: "notes",
+} as const;
+
+export function mapEntityToInitialFormState(
+    diseaseOutbreakEventWithOptions: DiseaseOutbreakEventWithOptions
+): FormState {
+    const { diseaseOutbreakEvent, options } = diseaseOutbreakEventWithOptions;
+    const {
+        teamMembers,
+        hazardTypes,
+        mainSyndromes,
+        suspectedDiseases,
+        notificationSources,
+        incidentStatus,
+    } = options;
+
+    const teamMemberOptions: UserOption[] = teamMembers.map(tm => ({
+        value: tm.username,
+        label: tm.name,
+        workPosition: tm.role?.name || "",
+        phone: tm.phone || "",
+        email: tm.email || "",
+        status: tm.status || "",
+        src: tm.photo?.toString(),
+        alt: tm.photo ? `Photo of ${tm.name}` : undefined,
+    }));
+
+    const hazardTypesOptions: PresentationOption[] = mapToPresentationOptions(hazardTypes);
+    const mainSyndromesOptions: PresentationOption[] = mapToPresentationOptions(mainSyndromes);
+    const suspectedDiseasesOptions: PresentationOption[] =
+        mapToPresentationOptions(suspectedDiseases);
+    const notificationSourcesOptions: PresentationOption[] =
+        mapToPresentationOptions(notificationSources);
+    const incidentStatusOptions: PresentationOption[] = mapToPresentationOptions(incidentStatus);
+
+    return {
```

Comment:
can we not generate this FormState by iterating over Object.Keys(diseaseOutbreakEvent) ? Maybe for future PRs?

Reply:
Adding a TODO for this